coreneuron/utils/randoms/nrnran123.cu

```diff
@@ -5,99 +5,107 @@
 # See top-level LICENSE file for details.
 # =============================================================================.
 */
+#include "coreneuron/utils/randoms/nrnran123.h"
+
+#ifdef _OPENMP
+#include "coreneuron/utils/nrnmutdec.h"
+#endif
 
 #include <stdio.h>
 #include <stdlib.h>
 #include <math.h>
-#include "coreneuron/utils/randoms/nrnran123.h"
-namespace coreneuron {
-/* global data structure per process */
-__device__ static const double SHIFT32 = 1.0 / 4294967297.0; /* 1/(2^32 + 1) */
-__device__ static philox4x32_key_t k = {{0}};
-__device__ static unsigned int instance_count_ = 0;
-__device__ size_t nrnran123_instance_count() {
-    return instance_count_;
-}
 
-__device__ size_t nrnran123_state_size() {
-    return sizeof(nrnran123_State);
-}
+#include <atomic>
```

Comment:
As mentioned during discussion, we should consider interoperability with OpenMP threading. For now, we can use MUTLOCK.

Reply:
I dropped `<atomic>` and now use the `OMP_Mutex` wrapper we already have.